### PR TITLE
automate marketplace image

### PR DESCRIPTION
**Pull request by sdelano**
/cc @ryancragun 

TODO:
* [x] upgrade recipe for delivery doesn't work - the ELK stack fails to start correctly
* [x] use randomized password for postgres
* [x] use randomized password for chef-server delivery user
* [x] store automate creds in a predetermined location?
* [x] use hostname helper to configure hostname
* [ ] install delivery from stable channel
* [x] use randomized passwords for admin and builder users from `chef-marketplace-ctl reconfigure`
* [ ] `chef-marketplace-ctl reconfigure` doesn't reconfigure delivery or chef-server if needed
* [x] biscotti fixes
  * render passwords so that the `marketplace` user can read them
  * display them when the cookie works
  * put it in its own `/credentials` route so that we don't need the nginx lua auth stuff in automate
* [ ] clean up `/etc/chef-marketplace/chef-marketplace-secrets.json` during prepare_for_publishing
* [x] deep_merge the secrets hash
* [x] get rid of nginx templates and use `chef_server_proxy` attribute
* [x] remove writing of delivery.license file

### CODE REVIEW

File: files/chef-marketplace-cookbooks/chef-marketplace/libraries/helpers.rb [hunk 1]
```diff
@@ -328,6 +360,6 @@ def compliance_state_directories
 if defined?(Chef)
   Chef::Recipe.send(:include, Marketplace::Helpers)
   Chef::Provider.send(:include, Marketplace::Helpers)
-  Chef::Resource.send(:include, Marketplace::Helpers)
+  Chef::Resource.send(:include, Marketplace::Helperso)
```
Comment (ryancragun) [inline files/chef-marketplace-cookbooks/chef-marketplace/libraries/helpers.rb]: Looks like a typo to me

File: files/chef-marketplace-cookbooks/chef-marketplace/templates/default/delivery.rb.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+delivery_fqdn "marketplace.vm"
```
Comment (ryancragun) [inline files/chef-marketplace-cookbooks/chef-marketplace/templates/default/delivery.rb.erb]: Not so sure we should be hardcoding this here.. Should probably use the FQDN helper

**Comment by ryancragun** [comment]
💄 🐷 

🎱 :shipit:

File: files/chef-marketplace-cookbooks/chef-marketplace/templates/default/delivery.rb.erb (hunk 2 above)
Comment (sdelano) [inline files/chef-marketplace-cookbooks/chef-marketplace/templates/default/delivery.rb.erb]: 👍

File: files/chef-marketplace-cookbooks/chef-marketplace/recipes/_config_secrets.rb [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
 
-    file secrets_file do
-      content Chef::JSONCompat.to_json_pretty(new_secrets)
-      sensitive true
-      action :create
-    end
+# Hash#merge will override any values in the original hash with values that
+# come from the input, so all values coming from the file on disk will be
+# overwrite the random data
+new_secrets = default_secrets.merge(secrets_from_file)
```
Comment (ryancragun) [inline files/chef-marketplace-cookbooks/chef-marketplace/recipes/_config_secrets.rb]: You sure? Don't nested hashes need to deep merged?

Comment (sdelano) [inline files/chef-marketplace-cookbooks/chef-marketplace/recipes/_config_secrets.rb]: bah..

Comment (sdelano) [inline files/chef-marketplace-cookbooks/chef-marketplace/recipes/_config_secrets.rb]: you're right, I was thinking I was fancy avoiding `deep_merge` but I'm just dumb

File: files/chef-marketplace-cookbooks/chef-marketplace/templates/default/nginx-chef-internal.conf.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#
+# chef server endpoints proxied by Delivery's nginx
```
Comment (ryancragun) [inline files/chef-marketplace-cookbooks/chef-marketplace/templates/default/nginx-chef-internal.conf.erb]: `delivery['chef_server_proxy'] = true` should create the upstreams for the chef server for you.

**Pull request merged**